### PR TITLE
`pytorch_lightning.loops` file structure: group by dataloader, epoch, and batch loop 

### PR DESCRIPTION
**Pull request by awaelchli**
## What does this PR do?

part of #7938 

The current structure is the following:

```
loops/
    dataloader/
        dataloader_loop.py
        evaluation_dataloader_loop.py
        prediction_dataloader_loop.py
    base.py
    fit_loop.py
    training_epoch_loop.py
    evaluation_epoch_loop.py
    prediction_epoch_loop.py
    training_batch_loop.py
```

The new proposed file structure:

```
loops/
    dataloader/
        dataloader_loop.py
        evaluation_loop.py
        prediction_loop.py
    epoch/
        training_epoch_loop.py
        evaluation_epoch_loop.py
        prediction_epoch_loop.py

    batch/
        training_batch_loop.py
        
    base.py
    fit_loop.py
```
No deprecations required. The new loops were introduced in master only so far and are for internal use currently.

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
I made sure I had fun coding 🙃

### CODE REVIEW

**Review by tchaton — APPROVED** [review]
LGTM !

**Review by justusschock — APPROVED** [review]

File: pytorch_lightning/loops/__init__.py [hunk 1]
```diff
@@ -13,8 +13,11 @@
 # limitations under the License.
 
 from pytorch_lightning.loops.base import Loop  # noqa: F401
+from pytorch_lightning.loops.batch.training_batch_loop import TrainingBatchLoop  # noqa: F401
 from pytorch_lightning.loops.dataloader.dataloader_loop import DataLoaderLoop  # noqa: F401
-from pytorch_lightning.loops.dataloader.evaluation_dataloader_loop import EvaluationDataLoaderLoop  # noqa: F401
+from pytorch_lightning.loops.dataloader.evaluation_loop import EvaluationLoop  # noqa: F401
+from pytorch_lightning.loops.dataloader.prediction_loop import PredictionLoop  # noqa: F401
```
Comment (carmocca) [inline pytorch_lightning/loops/__init__.py:19]: I prefer chaining `__init__` imports:

```python
from pytorch_lightning.loops.dataloader import DataLoaderLoop, EvaluationLoop, PredictionLoop # noqa: F401
```

File: pytorch_lightning/loops/__init__.py [hunk 2]
```diff
@@ -13,8 +13,11 @@
 # limitations under the License.
 
 from pytorch_lightning.loops.base import Loop  # noqa: F401
+from pytorch_lightning.loops.batch.training_batch_loop import TrainingBatchLoop  # noqa: F401
```
Comment (carmocca) [inline pytorch_lightning/loops/__init__.py:16]:
```suggestion
from pytorch_lightning.loops.batch import TrainingBatchLoop  # noqa: F401
```

File: pytorch_lightning/loops/__init__.py [hunk 3]
```diff
@@ -13,8 +13,11 @@
...
 from pytorch_lightning.loops.dataloader.dataloader_loop import DataLoaderLoop  # noqa: F401
-from pytorch_lightning.loops.dataloader.evaluation_dataloader_loop import EvaluationDataLoaderLoop  # noqa: F401
+from pytorch_lightning.loops.dataloader.evaluation_loop import EvaluationLoop  # noqa: F401
+from pytorch_lightning.loops.dataloader.prediction_loop import PredictionLoop  # noqa: F401
+from pytorch_lightning.loops.epoch.evaluation_epoch_loop import EvaluationEpochLoop  # noqa: F401
+from pytorch_lightning.loops.epoch.prediction_epoch_loop import PredictionEpochLoop  # noqa: F401
+from pytorch_lightning.loops.epoch.training_epoch_loop import TrainingEpochLoop  # noqa: F401
```
Comment (carmocca) [inline pytorch_lightning/loops/__init__.py:22]:
```suggestion
from pytorch_lightning.loops.epoch import EvaluationEpochLoop, PredictionEpochLoop, TrainingEpochLoop  # noqa: F401
```

File: pytorch_lightning/loops/fit_loop.py [hunk 4]
```diff
@@ -51,13 +51,13 @@ def __init__(
         super().__init__()
         self.max_epochs = 1000 if (max_epochs is None and max_steps is None) else max_epochs
         self.min_epochs = 1 if (min_epochs is None and min_steps is None) else min_epochs
-        self.training_loop = TrainingEpochLoop(min_steps, max_steps)
-        self.validation_loop = EvaluationDataLoaderLoop()
+        self.epoch_loop = TrainingEpochLoop(min_steps, max_steps)
```
Comment (carmocca) [inline pytorch_lightning/loops/fit_loop.py:54]: There's a conflict here with progress tracking.

The progress dataclass for fit has `train`  and `val` attributes which then have epoch/batch.

https://github.com/PyTorchLightning/pytorch-lightning/blob/8bd7b1bdd7d3f723822e78908033cf0a6743713a/pytorch_lightning/trainer/progress.py#L165-L166

The previous name did fit this structure but the new one doesn't

**Review by carmocca — DISMISSED** [review]

File: pytorch_lightning/loops/__init__.py (hunk 1 above)
Comment (awaelchli) [reply pytorch_lightning/loops/__init__.py:19]: sure, I just relied on isort to do the right thing.

Comment (carmocca) [reply pytorch_lightning/loops/__init__.py:19]: I don't think isort is that smart. It just checks the order and grouping but doesn't modify the import path

File: pytorch_lightning/loops/fit_loop.py (hunk 4 above)
Comment (awaelchli) [reply pytorch_lightning/loops/fit_loop.py:54]: it was suggested to me in the other PRs that `self.validation_loop.epoch_loop` is clearer to understand than `self.validation_loop.validation_loop`, so this was changed and I was following this logic. Let's brainstorm tomorrow how we want to make names consistent between progress tracking, loops, and rest of code base.

File: pytorch_lightning/loops/fit_loop.py [hunk 5]
```diff
@@ -51,15 +51,15 @@ def __init__(
...
+        self.val_loop = EvaluationLoop()
```
Comment (carmocca) [inline pytorch_lightning/loops/fit_loop.py:55]: Still need to move this inside the epoch loop as we discussed.

Or do you want to do that in a separate PR?

Comment (awaelchli) [reply pytorch_lightning/loops/fit_loop.py:55]: yes, I have it already implemented locally,
there is going to be a circular import due to that, so I would like to discuss the resolution of that in a separate PR.

**Review by carmocca — APPROVED** [review]
